spec/models/manageiq/providers/azure/cloud_manager/refresher_spec.rb

```diff
@@ -56,28 +63,54 @@ def assert_table_counts
     OperatingSystem.count.should eql(9)
     Relationship.count.should eql(0)
     MiqQueue.count.should eql(13)
-    OrchestrationTemplate.count.should eql(3)
+    OrchestrationTemplate.count.should eql(2)
     OrchestrationStack.count.should eql(7)
     OrchestrationStackParameter.count.should eql(72)
     OrchestrationStackOutput.count.should eql(5)
     OrchestrationStackResource.count.should eql(32)
+    SecurityGroup.count.should eql(9)
   end
 
   def assert_ems
     @ems.should have_attributes(
       :api_version => nil,
-      :uid_ems     => "a50f9983-d1a2-4a8d-be7d-123456789012"
+      :uid_ems     => @tenantid
     )
     @ems.flavors.size.should eql(33)
     @ems.availability_zones.size.should eql(1)
     @ems.vms_and_templates.size.should eql(13)
     @ems.vms.size.should eql(9)
+    @ems.security_groups.size.should eql(9)
     @ems.miq_templates.size.should eq(4)
 
     @ems.orchestration_stacks.size.should eql(7)
     @ems.direct_orchestration_stacks.size.should eql(6)
   end
 
+  def assert_specific_security_group
+    @sg = ManageIQ::Providers::Azure::CloudManager::SecurityGroup.where(:name => "Chef-Prod").first
+
+    @sg.should have_attributes(
+      :name        => "Chef-Prod",
+      :description => "Chef-Prod - Chef-Prod - eastus"
+    )
+
+    expected_firewall_rules = [
+      {:host_protocol => "TCP", :direction => "Inbound", :port => 22,  :end_port => 22,  :source_ip_range => "*"},
+      {:host_protocol => "TCP", :direction => "Inbound", :port => 80,  :end_port => 80,  :source_ip_range => "*"},
+      {:host_protocol => "TCP", :direction => "Inbound", :port => 443, :end_port => 443, :source_ip_range => "*"}
+    ]
+
+    @sg.firewall_rules.size.should == 3
+
+    @sg.firewall_rules
+      .order(:host_protocol, :direction, :port, :end_port, :source_ip_range, :source_security_group_id)
+      .zip(expected_firewall_rules)
+      .each do |actual, expected|
+        actual.should have_attributes(expected)
+      end
+  end
+
   def assert_specific_flavor
     @flavor = ManageIQ::Providers::Azure::CloudManager::Flavor.where(:name => "Standard_A0").first
     @flavor.should have_attributes(
```

Comment:
@djberg96 
Looks like we need to test the security_groups property on instances both in the assert_specific_vm_powered_on and assert_specific_vm_powered_off methods.